core/modules/replica/Request.cc

```diff
@@ -0,0 +1,270 @@
+                    "incomplete implementation of method Request::state2string(ExtendedState)");
+}
+
+std::string Request::state2string(State state,
+                                  ExtendedState extendedState) {
+    return state2string(state) + "::" + state2string(extendedState);
+}
+
+std::string Request::state2string(State state,
+                                  ExtendedState extendedState,
+                                  replica::ExtendedCompletionStatus serverStatus) {
+    return state2string(state, extendedState) + "::" + replica::status2string(serverStatus);
+}
+
+Request::Request(ServiceProvider::pointer const& serviceProvider,
+                 boost::asio::io_service& io_service,
+                 std::string const& type,
+                 std::string const& worker,
+                 int  priority,
+                 bool keepTracking,
+                 bool allowDuplicate)
+    :   _serviceProvider(serviceProvider),
+        _type(type),
+        _id(Generators::uniqueId()),
+        _worker(worker),
+        _priority(priority),
+        _keepTracking(keepTracking),
+        _allowDuplicate(allowDuplicate),
+        _state(CREATED),
+        _extendedState(NONE),
+        _extendedServerStatus(ExtendedCompletionStatus::EXT_STATUS_NONE),
+        _performance(),
+        _bufferPtr(new ProtocolBuffer(serviceProvider->config()->requestBufferSizeBytes())),
+        _workerInfo(serviceProvider->config()->workerInfo(worker)),
+        _timerIvalSec(serviceProvider->config()->retryTimeoutSec()),
+        _timer(io_service),
+        _requestExpirationIvalSec(serviceProvider->config()->controllerRequestTimeoutSec()),
+        _requestExpirationTimer(io_service) {
+
+        _serviceProvider->assertWorkerIsValid(worker);
+}
+std::string Request::context() const {
+    return "REQUEST " + id() + "  " + type() +
+           "  " + state2string(state(), extendedState()) +
+           "::" + replica::status2string(extendedServerStatus()) + "  ";
+}
+
+std::string const& Request::remoteId() const {
+    return _duplicateRequestId.empty() ? _id : _duplicateRequestId;
+}
+
+void Request::start(std::shared_ptr<Controller> const& controller,
+                    std::string const& jobId,
+                    unsigned int requestExpirationIvalSec) {
+
+    LOCK_GUARD;
+
+    assertState(CREATED);
+
+    // Change the expiration ival if requested
+    if (requestExpirationIvalSec) {
+        _requestExpirationIvalSec = requestExpirationIvalSec;
+    }
+    LOGS(_log, LOG_LVL_DEBUG, context() << "start  _requestExpirationIvalSec: "
+         << _requestExpirationIvalSec);
+
+    // Build optional associaitons with the corresponding Controller and the job
+    //
+    // NOTE: this is done only once, the first time a non-trivial value
+    // of each parameter is presented to the method.
+
+    if (not _controller    and     controller)    { _controller = controller; }
+    if (    _jobId.empty() and not jobId.empty()) { _jobId      = jobId; }

```

Comment:
in Request you've used the one-line form of the if statement to do a quick return based on some precondition data which I think is ok, but this goes a little too far. Please separate these so the body is on the next line.

Reply:
I see our style guide allows one line if statements, which surprises me but ok. So you can disregard this comment. You're supposed to omit the braces, I see the comments about that on Andy's comment above. I'll add my editorial there.

Reply:
I omitted braces.